utils/post.js

```diff
@@ -310,39 +310,50 @@ function modifyReactionsPost(post, isAnonimous = true) {
   return newPost;
 }
 
-async function filterFeeds(userId, feeds = [], feed_id = null) {
-  const newResult = [];
-
-  for (const item of feeds || []) {
-    const now = moment().valueOf();
-    const dateExpired = moment(item?.expired_at).valueOf();
-
-    if (dateExpired < now) {
-      if (feed_id) {
-        deleteActivityFromUserFeed('topic', feed_id, item.id);
-      }
-      continue;
+const isExpiredPost = (item, feed_id) => {
+  let result = false;
+  const now = moment().valueOf();
+  const dateExpired = moment(item?.expired_at).valueOf();
+
+  if (dateExpired < now) {
+    if (feed_id) {
+      deleteActivityFromUserFeed('topic', feed_id, item.id);
     }
-    let newItem = {...item};
+    result = true;
+  }
+  return result;
+};
 
-    if (newItem.anonimity) {
-      newItem.actor = {};
-      newItem.to = [];
-      newItem.origin = null;
-      newItem.object = '';
-    }
+const modifyNewItemAnonymity = (newItem) => {
+  if (newItem.anonimity) {
+    newItem.actor = {};
+    newItem.to = [];
+    newItem.origin = null;
+    newItem.object = '';
+  }
+  return newItem;
+};
 
-    newItem = modifyReactionsPost(newItem, newItem.anonimity);
+async function filterFeeds(userId, feeds = [], feed_id = null, threshold = null) {
+  const newResult = feeds
+    .filter(async (item) => {
+      const expired = await isExpiredPost(item, feed_id);
+      return !(expired || (threshold && (item.final_score || 0) < threshold));
+    })
+    .map(async (item) => {
+      let newItem = {...item};
 
-    const isValidPollPost = item.verb === POST_VERB_POLL && item?.polls?.length > 0;
+      newItem = modifyNewItemAnonymity(newItem);
+      newItem = modifyReactionsPost(newItem, newItem.anonimity);
 
-    if (isValidPollPost) newItem = await modifyPollPostObject(userId, newItem);
-    else newItem = await modifyPostLinkPost(DomainPage, newItem);
+      const isValidPollPost = item.verb === POST_VERB_POLL && item?.polls?.length > 0;
 
-    newResult.push(newItem);
-  }
+      if (isValidPollPost) newItem = await modifyPollPostObject(userId, newItem);
+      else newItem = await modifyPostLinkPost(DomainPage, newItem);
 
-  return newResult;
+      return newItem;
+    });
+  return Promise.all(newResult);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The filter function in JavaScript is synchronous and does not support async operations. Using async inside filter will not work as expected. Instead, you should use a combination of `Promise.all()` and `Array.prototype.map()` to achieve the desired result. Also, consider using the new `deleteExpiredPostFromFeed` function here.
```diff
async function filterFeeds(userId, feeds = [], feed_id = null, threshold = null) {
  const results = await Promise.all(feeds.map(async (item) => {
    const expired = isExpiredPost(item);
    if (expired) {
      deleteExpiredPostFromFeed(item, feed_id);
      return null;
    }
    if (threshold && (item.final_score || 0) < threshold) {
      return null;
    }
    let newItem = {...item};
 
    newItem = modifyNewItemAnonymity(newItem);
    newItem = modifyReactionsPost(newItem, newItem.anonimity);
 
    const isValidPollPost = item.verb === POST_VERB_POLL && item?.polls?.length > 0;
 
    if (isValidPollPost) newItem = await modifyPollPostObject(userId, newItem);
    else newItem = await modifyPostLinkPost(DomainPage, newItem);
 
    return newItem;
  }));
  return results.filter(item => item !== null);
}
```
This change will ensure that the filter operation works as expected with async operations and also improves the readability of the code by separating concerns into different functions.


<!-- This is an auto-generated comment by OSS CodeRabbit -->